courses/programming-in-haskell/pih-exercises/test/Chapter05Spec.hs

```diff
@@ -0,0 +1,29 @@
+-- | Specification for the exercises of Chapter 04.
+
+module Chapter05Spec where
+
+import Chapter05 (pyths, perfects, scalarproduct)
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+spec :: Spec
+spec = do
```

Comment:
Here you can remove all the type bindings, since there is no ambiguity, i.e. `pyths :: Int -> [(Int,Int,Int)]` which means that the compiler infers that if you apply `pyths 4` then `4` must be an `Int`. The same holds for the other type instantiations. I'll modify this in the master branch.